src/server/registry/catalog/catalog.go

```diff
@@ -14,19 +14,109 @@
 
 package catalog
 
-import "net/http"
+import (
+	"encoding/json"
+	ierror "github.com/goharbor/harbor/src/internal/error"
+	"github.com/goharbor/harbor/src/pkg/repository"
+	"github.com/goharbor/harbor/src/server/registry/error"
+	reg_error "github.com/goharbor/harbor/src/server/registry/error"
+	"github.com/goharbor/harbor/src/server/registry/util"
+	"net/http"
+	"sort"
+	"strconv"
+)
 
 // NewHandler returns the handler to handler catalog request
-func NewHandler() http.Handler {
-	return &handler{}
+func NewHandler(repoMgr repository.Manager) http.Handler {
+	return &handler{
+		repoMgr: repoMgr,
+	}
 }
 
-type handler struct{}
+type handler struct {
+	repoMgr repository.Manager
+}
 
+// ServeHTTP ...
 func (h *handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
-	// TODO implement
+	switch req.Method {
+	case http.MethodGet:
+		h.get(w, req)
+	}
+}
+
+func (h *handler) get(w http.ResponseWriter, req *http.Request) {
+	var withPagination = true
+
+	reqQ := req.URL.Query()
+	lastEntry := reqQ.Get("last")
+	maxEntries, err := strconv.Atoi(reqQ.Get("n"))
+	if err != nil || maxEntries < 0 {
+		err := ierror.New(err).WithCode(ierror.BadRequestCode).WithMessage("the N must be a positive int type")
+		reg_error.Handle(w, req, err)
+		return
+	}
+	if maxEntries == 0 {
+		withPagination = false
+	}
+
+	var repos []string
+	// get all repositories
+	total, repoRecords, err := h.repoMgr.List(req.Context(), nil)
+	if err != nil {
+		error.Handle(w, req, err)
+		return
+	}
+	if total <= 0 {
+		h.sendResponse(w, req, repos)
```

Comment:
The response data should be in the format of `catalogAPIResponse`

Reply:
yes, it is.